### ISSUE TITLE
API asset retrieval needs an evaluate and retry method (if it fails)

### ISSUE THREAD

**Issue by toconnell** [toconnell/kdm-manager#162]
Sometimes it just doesn't get results. Not sure why, but need to hack around it while I figure it out.

**Comment by toconnell** [comment]
```
[2016-11-27 22:30:24] INFO:     User 'XXX' created new survivor Punchy McFist [M] (583b5e808740d96cc7d24466) successfully.
[2016-11-27 22:30:25] WARNING:  ['XXX' (583a60a38740d930da681ea2)] asset type 'game_assets' not found in API asset for 'Arizona' (583a60d08740d9310faeadd4)
[2016-11-27 22:30:25] DEBUG:    ['XXX' (583a60a38740d930da681ea2)] available API asset_types for 'Arizona' (583a60d08740d9310faeadd4): []
[2016-11-27 22:30:25] WARNING:  ['XXX' (583a60a38740d930da681ea2)] api_asset key 'cursed_items' does not exist for Punchy McFist [M]!
[2016-11-27 22:30:25] DEBUG:    ['XXX' (583a60a38740d930da681ea2)] available API asset keys for Punchy McFist [M]: []
```

**Comment by toconnell** [comment]
This should all be sorted out: the release that's about to go out (1.52.1194)
